plutus-core/plutus-core/src/PlutusCore/Builtin/Runtime.hs

```diff
@@ -131,33 +67,28 @@ data UnliftingMode
     | UnliftingDeferred
 
 -- | A 'BuiltinRuntimeOptions' is a precursor to 'BuiltinRuntime'. One gets the latter from the
--- former by choosing the runtime denotation of the builtin (either '_broImmediateF' for immediate
--- unlifting or '_broDeferredF' for deferred unlifting, see 'UnliftingMode' for details) and by
--- instantiating '_broToExF' with a cost model to get the costing function for the builtin.
---
--- The runtime denotations are lazy, so that we don't need to worry about a builtin being bottom
--- (happens in tests). The production path is not affected by that, since 'BuiltinRuntimeOptions'
--- doesn't survive optimization.
-data BuiltinRuntimeOptions n val cost = BuiltinRuntimeOptions
-    { _broRuntimeScheme :: RuntimeScheme n
-    , _broImmediateF    :: ~(ToRuntimeDenotationType val n)
-    , _broDeferredF     :: ~(ToRuntimeDenotationType val n)
-    , _broToExF         :: cost -> ToCostingType n
+-- former by applying a function returning the runtime denotation of the builtin (either
+-- '_broImmediateF' for immediate unlifting or '_broDeferredF' for deferred unlifting, see
+-- 'UnliftingMode' for details) to a cost model.
+data BuiltinRuntimeOptions val cost = BuiltinRuntimeOptions
+    { _broImmediateF :: cost -> BuiltinRuntime val
+    , _broDeferredF  :: cost -> BuiltinRuntime val
     }
 
 -- | Convert a 'BuiltinRuntimeOptions' to a 'BuiltinRuntime' given an 'UnliftingMode' and a cost
 -- model.
 fromBuiltinRuntimeOptions
-    :: UnliftingMode -> cost -> BuiltinRuntimeOptions n val cost -> BuiltinRuntime val
-fromBuiltinRuntimeOptions unlMode cost (BuiltinRuntimeOptions sch immF defF toExF) =
-    BuiltinRuntime sch f $ toExF cost where
-        f = case unlMode of
-                UnliftingImmediate -> immF
-                UnliftingDeferred  -> defF
+    :: UnliftingMode -> cost -> BuiltinRuntimeOptions val cost -> BuiltinRuntime val
+fromBuiltinRuntimeOptions unlMode cost (BuiltinRuntimeOptions immF defF) =
+    case unlMode of
+        UnliftingImmediate -> immF cost
+        UnliftingDeferred  -> defF cost
 {-# INLINE fromBuiltinRuntimeOptions #-}
 
 instance NFData (BuiltinRuntime val) where
-    rnf (BuiltinRuntime rs f exF) = rnf rs `seq` f `seq` rwhnf exF
+    -- 'BuiltinRuntime' is strict (verified by the 'NoThunks' tests), hence we only need to force
+    -- this to WHNF to get it forced to NF.
+    rnf = rwhnf
 
 -- | A 'BuiltinRuntime' for each builtin from a set of builtins.
 newtype BuiltinsRuntime fun val = BuiltinsRuntime
```

Comment:
Irrelevant question: in the very final line of the file it has 
```
Just bri -> pure bri
```
What does `bri` stand for?

Reply:
Builtin runtime I-don't-know-what. Some legacy name, I'll fix it.